### PR TITLE
Feat/deposit stake

### LINKED ISSUE #77: feat: deposit stake

### PR DESCRIPTION
**Pull request by billythedummy**
Closes #77

### CODE REVIEW

File: src/socean.ts [hunk 1]
```diff
@@ -166,6 +170,144 @@ export class Socean {
...
+   *                     Must be active and delegated to a validator in the stake pool.
+   *                     The entire stake account is deposited. To deposit only a portion of it, prefix this instruction
+   *                     with a stake split instruction and deposit the stake account that's split off instead.
+   * @param referrerPoolTokenAccount PublicKey of a scnSOL token account of the referrer for this deposit
+   * @param confirmOptions transaction confirm options for each transaction
+   * @returns the transaction signatures of the transactions sent and confirmed
+   * @throws RpcError
+   * @throws AccountDoesNotExistError if stake pool or main validator stake account for `stakeAccount`'s validator does not exist
+   * @throws WalletPublicKeyUnavailableError
+   * @throws StakeAccountToDepositInvalidError if stake account to deposit does not meet deposit requirements
```
Comment (f8122dac91) [inline src/socean.ts:187]: maybe we can add a small helpful hint describing the deposit requirements for the people debugging?

**Review by f8122dac91 — APPROVED** [review]
Looks good to me. Added one comment to consider for the doc.

File: src/socean.ts (hunk 1 above)
Comment (billythedummy) [reply src/socean.ts:187]: see the doc comment for `@param stakeAccount`

Comment (f8122dac91) [reply src/socean.ts:187]: Fair 👍

**Pull request merged**